examples/cpp/function-execution/main.cpp

```diff
@@ -38,147 +34,114 @@
+  auto pool = cache.getPoolManager()
+      .createFactory()
+      .addServer("localhost", EXAMPLE_SERVER_PORT)
+      .create("pool");
+}
 
-const int EXAMPLE_ITEM_COUNT = 34;
+std::shared_ptr<Region> createRegion(Cache& cache) {
+  auto regionFactory = cache.createRegionFactory(RegionShortcut::PROXY);
+  auto region = regionFactory.setPoolName("pool").create("partition_region");
 
-// The Execute Function QuickStart example.
-int main(int argc, char** argv) {
-  try {
-    // Create CacheFactory using the settings from the geode.properties file by
-    // default.
-    auto cache = CacheFactory()
-        .set("log-level", "none")
-        .create();
-
-    std::cout << "Created CacheFactory\n";
-
-    auto pool = cache.getPoolManager()
-        .createFactory()
-        .setSubscriptionEnabled(true)
-        .addServer("localhost", 50505)
-        .addServer("localhost", 40404)
-        .create("pool");
-
-    // Create the example Region Programmatically
-    auto regionFactory = cache.createRegionFactory(RegionShortcut::PROXY);
-    auto regPtr0 = regionFactory.setPoolName("pool").create("partition_region");
-
-    std::cout << "Created the Region\n";
-
-    regPtr0->registerAllKeys();
-    char buf[128];
-
-    auto resultList = CacheableVector::create();
-    for (int i = 0; i < EXAMPLE_ITEM_COUNT; i++) {
-      sprintf(buf, "VALUE--%d", i);
-      auto value(CacheableString::create(buf));
-
-      sprintf(buf, "KEY--%d", i);
-      auto key = CacheableKey::create(buf);
-      regPtr0->put(key, value);
-    }
+  return region;
+}
 
-    auto routingObj = CacheableVector::create();
-    for (int i = 1; i < EXAMPLE_ITEM_COUNT; i+=2) {
-      sprintf(buf, "KEY--%d", i);
-      auto key = CacheableKey::create(buf);
-      routingObj->push_back(key);
-    }
+void populateRegion(const std::shared_ptr<Region>& region) {
+  for (int i = 0; i < keys.size(); i++) {
+    region->put(keys[i], values[i]);
+  }
+}
 
-    std::cout << "test data independent function with result on one server\n";
-    auto exc = FunctionService::onServer(regPtr0->getRegionService());
-    if(auto executeFunctionResult = exc.withArgs(routingObj).execute(getFuncIName)->getResult()) {
-      for (auto &arrayList: *executeFunctionResult) {
-        for (auto &cachedString: *std::dynamic_pointer_cast<CacheableArrayList>(arrayList)) {
-          resultList->push_back(cachedString);
-        }
-      }
-      sprintf(buf, "get: result count = %lu\n", resultList->size());
-      std::cout << buf;
-      int i = 0;
-      for (auto &cachedString: *resultList) {
-        sprintf(
-            buf, "get result[%d]=%s\n", i,
-            std::dynamic_pointer_cast<CacheableString>(cachedString)->value().c_str());
-        std::cout << buf;
-        ++i;
+std::shared_ptr<CacheableVector> populateArguments() {
+  auto arguments = CacheableVector::create();
+  for (int i = 0; i < keys.size(); i++) {
+    arguments->push_back(CacheableKey::create(keys[i]));
```

Comment:
I think you can send vector of `std::string`. You should try it. If not you should open a ticket to support it.

Reply:
At first blush it doesn't appear to work.  GEODE-6004 filed for tracking.